src/coreclr/jit/utils.cpp

```diff
@@ -999,6 +1000,90 @@ void Histogram::record(unsigned size)
     m_counts[i]++;
 }
 
+void NodeCounts::dump(FILE* output)
+{
+    struct Entry
+    {
+        genTreeOps oper;
+        unsigned   count;
+    };
+
+    Entry sorted[GT_COUNT];
+    for (int i = 0; i < GT_COUNT; i++)
+    {
+        sorted[i].oper  = static_cast<genTreeOps>(i);
+        sorted[i].count = m_counts[i];
+    }
+
+    jitstd::sort(sorted, sorted + ArrLen(sorted), [](const Entry& lhs, const Entry& rhs) {
+        if (lhs.count > rhs.count)
+        {
+            return true;
+        }
+
+        if (lhs.count < rhs.count)
+        {
+            return false;
+        }
+
+        return static_cast<unsigned>(lhs.oper) < static_cast<unsigned>(rhs.oper);
+    });
+
+    for (const Entry& entry : sorted)
+    {
+        if (entry.count == 0)
+        {
+            break;
+        }
+
+        fprintf(output, "%-20s : %7u\n", GenTree::OpName(entry.oper), entry.count);
+    }
+}
+
+void NodeCounts::record(genTreeOps oper)
+{
+    assert(oper < GT_COUNT);
+    m_counts[oper]++;
```

Comment:
Should this do an interlocked add?

Reply:
Maybe that would depend on whether this class is instantiated as global, or as a member of Compiler?

Reply:
Yeah, for good measure it should be interlocked add. `NodeCounts` is meant to be used as a static field (like `Histogram`) so you can collect stats for multiple compilations, so it can be racing with other threads, although I expect we mainly use it through SPMI where it won't due to the way parallelization works there.

Looks like `Histogram` should also be changed to use an interlocked add.